provisioning/device/src/polling_state_machine.ts

```diff
@@ -145,6 +144,12 @@ export class  PollingStateMachine extends EventEmitter {
             this._pollingTimer = null;
             this._fsm.transition('cancelling', err, result, response, callback);
           },
+          /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_034: [ If `disconnect` is called while the state machine is waiting to poll, the current operation shall be cancelled and the transport shall be disconnected. ] */
+          disconnect: (callback) => {
+            this._fsm.handle('cancel', null, null, null, (err) => {
+              this._fsm.transition('disconnecting', callback);
```

Comment:
>transition [](start = 24, length = 10)

same here - handle('disconnect', callback) instead of transition (and in other places too)